asterius/src/Asterius/Types/SymbolMap.hs

```diff
@@ -75,6 +76,9 @@ newtype SymbolMap a = SymbolMap (IM.IntMap (EntitySymbol, a))
   deriving newtype (Eq, Semigroup, Monoid)
   deriving stock (Data)
 
+instance NFData a => NFData (SymbolMap a) where
```

Comment:
IMO it would be nice to put it to `deriving newtype`. Same applies for `SymbolSet`. Not for extra performance, just that we can save a bit of boilerplate code.